api/v1alpha3/azuremachine_types.go

```diff
@@ -52,11 +52,21 @@ type AzureMachineSpec struct {
 
 	// Identity is the type of identity used for the virtual machine.
 	// The type 'SystemAssigned' is an implicitly created identity
-	// The generated identity will be assigned a Subscription contributor role
+	// The generated identity will be assigned a Subscription contributor role.
+	// The type 'UserAssigned' is a standalone Azure resource provided by the user
+	// and assigned to the VM
 	// +kubebuilder:default=None
 	// +optional
 	Identity VMIdentity `json:"identity,omitempty"`
 
+	// UserAssignedIdentities is a list of standalone Azure identities provided by the user
+	// The lifecycle of a user-assigned identity is managed separately from the lifecycle of
+	// the AzureMachine.
+	// The format of an identity is:
+	// '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}'
+	// +optional
+	UserAssignedIdentities []string `json:"userAssignedIdentities,omitempty"`
```

Comment:
I think we were hoping to make `UserAssignedIdentities` a struct with a `ProviderID string` field. The string for `ProviderID` should be formatted like other `ProviderID`s (`azure:////...`).

https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/312#issuecomment-628266143

Reply:
ok, I started doing that but didn't want to add a struct for one field, anyways I don't mind it, so I'll make the change

Reply:
Agreed. It's not for the one field, it's so that as the fields that are needed to allow for creation of identities are added, it would be a non-breaking changes. We could add fields to the struct to provide the data needed to create the identity.